From 608091fd2c161f2c9af2eb5b1af44929e6710bac Mon Sep 17 00:00:00 2001
From: Norbert Wesp <n.wesp@phytec.de>
Date: Tue, 28 Jan 2020 13:00:19 +0100
Subject: [PATCH] mesa: fix imx gallium driver

original commit message:
    Fix imx autotools build dependencies

    Since commit 41a0acd6 ("Switch imx to kmsro and remove the imx winsys")
    the imx-drm is covered by kmsro. During the switch some autotools
    dependencies are missed. Autotools are marked as deprecated but since
    yocto still uses them we need to cover this.

    Signed-off-by: Marco Felsch <m.felsch@pengutronix.de>

Signed-off-by: Norbert Wesp <n.wesp@phytec.de>
---
 configure.ac                           | 4 ++--
 src/gallium/drivers/kmsro/Automake.inc | 1 +
 2 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/configure.ac b/configure.ac
index bfb6b8c..c40a68a 100644
--- a/configure.ac
+++ b/configure.ac
@@ -2856,8 +2856,8 @@ AM_CONDITIONAL(HAVE_SWR_BUILTIN, test "x$HAVE_SWR_BUILTIN" = xyes)
 
 dnl We need to validate some needed dependencies for renderonly drivers.
 
-if test "x$HAVE_GALLIUM_VC4" != xyes -a "x$HAVE_GALLIUM_KMSRO" = xyes  ; then
-    AC_MSG_ERROR([Building with kmsro requires vc4])
+if test "x$HAVE_GALLIUM_VC4" != xyes -a "x$HAVE_GALLIUM_ETNAVIV" != xyes -a "x$HAVE_GALLIUM_KMSRO" = xyes  ; then
+    AC_MSG_ERROR([Building with kmsro requires vc4 or etnaviv])
 fi
 
 if test "x$HAVE_GALLIUM_NOUVEAU" != xyes -a "x$HAVE_GALLIUM_TEGRA" = xyes; then
diff --git a/src/gallium/drivers/kmsro/Automake.inc b/src/gallium/drivers/kmsro/Automake.inc
index d5961c9..ebed424 100644
--- a/src/gallium/drivers/kmsro/Automake.inc
+++ b/src/gallium/drivers/kmsro/Automake.inc
@@ -2,6 +2,7 @@ if HAVE_GALLIUM_KMSRO
 
 TARGET_DRIVERS += pl111
 TARGET_DRIVERS += hx8357d
+TARGET_DRIVERS += imx-drm
 TARGET_CPPFLAGS += -DGALLIUM_KMSRO
 TARGET_LIB_DEPS += \
     $(top_builddir)/src/gallium/winsys/kmsro/drm/libkmsrodrm.la \
-- 
2.7.4

